### ISSUE TITLE
Unable to boot into Ubuntu Touch, installation error (MultiRom & Dev Flash)

### ISSUE THREAD

**Issue by joelheaps** [ubports/ubuntu-touch-for-oneplus-one#32]
This may be a duplicate, but the other post didn't include any info.

**My OnePlus hangs when booting into Ubuntu Touch (utouch) from MultiRom, and cannot complete installation with the device flash method.**

Recovery version 2015-06-07 won't boot when installed from the custom MultiRom app.  I did a workaround, installing the 2016 recovery with the custom 32j MultiRom version, and a keysign error workaround from the OnePlus forum.  _The installer abrubtly reboots a little while after displaying "Updating ubuntu-***.tar.xz".  utouch appears in the MultiRom list, but every attempt to boot it fails._  I've tried with Stable versions 7 and 9.

Edit: I've now tried the device flash method from the ubports website, and the flash fails with that as well.  Phone reboots into Android (which should have been erased) when using the device flash method.

++ **It appears that the device flash successfully got as far as flashing the recovery, but the phone couldn't boot whatever recovery was flashed.**  Holding down volume down to enter recovery, the phone just hangs and reboots.  In the Ubuntu terminal on my PC, it just says _"Failed to enter recovery"_

Will provide more info on request.  Any help appreciated.

**Comment by prajwal-orion** [comment]
Hi i too have the same problem, the devel-proposed v208 and stable v10 both give error as keyring missing, it just doesnt work. 
Trying to try out your build but not able to :/

**Comment by joelheaps** [comment]
Yeah, the "device-signing" not found (or whatever it says) might not be the culprit.  I think that's characteristic of other Ubuntu Touch devices too.  I don't know for sure though.  I wish I could find a detailed log so I could see why the installation was failing.

The weird thing is it seems to do something similar when I use the full device-flash method too. Doesn't work.

**Comment by ghost** [comment]
I'm having the same issue. I've installed the latest cm12.1 snapshot. The latest AK kernel. I've installed the ub multirom manager and twrp version from here. I've tried installing a few different versions of Ubuntu. They seem to install correctly, although I do see the key ring thing and also notice a warning about swap file creation. But none of them have booted properly. Just gets stuck at the oneplus logo splash screen. I have injected the boot loader. I also tried switching to the lightning kernel, but that one wouldn't even boot my internal cm12. Any thoughts on what to try would be greatly appreciated.

**Comment by joelheaps** [comment]
Tried with the device flash method from ubports.  Device fails to reboot into recovery (is the recovery only compatible with some versions of this phone?).  I updated my original comment to reflect the same info.

**Comment by joelheaps** [comment]
Cleaned up and re-posted on the new Launchpad bug reporting page.
[https://bugs.launchpad.net/ubports-android/+bug/1556673](https://bugs.launchpad.net/ubports-android/+bug/1556673)

**Comment by binbot** [comment]
While I have not tried the multirom method, I can confirm device flashing does not work. the device fails to boot to recovery after the push.

**Comment by rdmarsee** [comment]
I may have found a possible solution. I too was unable to flash Ubuntu touch on my OPO, but I noticed many of the successful stories were from the KitKat era. I went back and flashed the 'cm-11-XNPH44S' build and suddenly the Utouch flash went without incident. I think the Ubuntu recovery just needs to be recompiled to support the more modern bootloaders on the devices of those of us who have updated to COS 12.x + Perhaps there is a different partition layout for the newer builds?
